### ISSUE TITLE
Does CodeQL CLI support CUDA C++ (.cu) scanning ?

### ISSUE THREAD

**Issue by mrragava** [github/codeql#12943]
Does CodeQL CLI support CUDA C++ (.cu) scanning ? 
If not, Is there a way for us to extend the language support in extractor ?

**Comment by hmakholm** [comment]
No, CodeQL does not support CUDA C/C++ extensions directly, and we do not have immediate plans for adding such support. Extending the C/C++ extractor to understand them would be a large undertaking (and one that needs to be done inside GitHub, since the extractor is not open-sourced).

However, there seems to be a good chance that the extractor will be able out of the box to intercept the modified C++ code that the `nvcc` driver hands off to a _host_ C++ compiler, and analyze _that_. The GPU parts of the code will not be analyzed that way, but that's probably a relatively unlikely place to find the kind of security risks that CodeQL looks for anyway.

(Beware that we have not actually tried this ourselves; we don't really have any hands-on CUDA experience on the relevant teams, so this suggestion is based on web searches to learn how it works).
